src/main/java/org/isf/medicalinventory/manager/MedicalInventoryRowManager.java

```diff
@@ -76,6 +88,70 @@ public void deleteMedicalInventoryRow(MedicalInventoryRow medicalInventoryRow) t
 	 * @throws OHServiceException
 	 */
 	public List<MedicalInventoryRow> getMedicalInventoryRowByInventoryId(int inventoryId) throws OHServiceException {
-		return iOoperation.getMedicalInventoryRowByInventoryId(inventoryId);
+		return ioOperation.getMedicalInventoryRowByInventoryId(inventoryId);
+	}
+
+	/**
+	 * Delete a list of inventory rows {@link MedicalInventoryRow}s
+	 *
+	 * @param inventoryRowsToDelete - the list of {@link MedicalInventoryRow}s
+	 * 
+	 * @throws OHServiceException
+	 */
+	@Transactional(rollbackOn = OHServiceException.class)
+	public void deleteMedicalInventoryRows(List<MedicalInventoryRow> inventoryRowsToDelete) throws OHServiceException {
+		for (MedicalInventoryRow invRow : inventoryRowsToDelete) {
+			Optional<MedicalInventoryRow> medInvRow = ioOperation.getMedicalInventoryRowById(invRow.getId());
+			if (medInvRow.isPresent()) {
+				MedicalInventoryRow invRowDelete = medInvRow.get();
+				if (invRowDelete.isNewLot()) {
+					this.deleteMedicalInventoryRow(invRowDelete);	
+					if (invRowDelete.getLot() != null) {
+						movStockInsertingManager.deleteLot(invRowDelete.getLot());
+					}
+				} else {
+					this.deleteMedicalInventoryRow(invRowDelete);
+				}
+			}	
+		}
+	}
+
+	/**
+	 * Return {@link MedicalInventoryRow} for passed param.
+	 *
+	 * @param invRowId - the MedicalInventoryRow Id.
+	 * @return the {@link MedicalInventoryRow} object.
+	 * @throws OHServiceException
+	 */
+	public MedicalInventoryRow getMedicalInventoryRowById(Integer invRowId) throws OHServiceException {
+		if (invRowId != null) {
+			Optional<MedicalInventoryRow> medInvRow = ioOperation.getMedicalInventoryRowById(invRowId);
+			if (medInvRow.isPresent()) {
+				return medInvRow.get();
+			}
+		}
+		
+		return null;
+	}
+	
+	/**
+	 * Verify if the object is valid for CRUD and return a list of errors, if any.
+	 *
+	 * @param medInventoryRow
+	 * @throws OHDataValidationException
+	 */
+	private void validationMedicalInventoryRow(MedicalInventoryRow medicalInventoryRow) throws OHDataValidationException {
```

Comment:
Doesn't other code in the project use `validate` for the first part of the method name instead of `validation`?